File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/04_quicksort/ts/05_quicksort.ts
function quicksort(arr: number[]): number[] {
    if (arr.length < 2) {
        return arr;
    }

    const pivot = arr[0];
    const less = arr.slice(1).filter(el => el <= pivot);
    const greater = arr.slice(1).filter(el => el > pivot);

    return quicksort(less).concat(pivot, quicksort(greater));
}

console.log(quicksort([10, 5, 2, 3]));

File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/__MY_OPRIGINAL_DS/_Extra-Practice/04_quicksort/julia/05_quick_sort.jl
using Test

function quick_sort(arr)
    if length(arr) < 2 return arr end
    pivot = arr[1]
    less = [i for i in arr[2:end] if i <= pivot]
    greater = [i for i in arr[2:end] if i > pivot]
    return vcat(quick_sort(less), [pivot], quick_sort(greater))
end

@test quick_sort([3,5,2,1,4]) == [1,2,3,4,5]
